app/src/main/java/com/epfl/beatlink/ui/profile/settings/Account.kt

```diff
@@ -161,3 +160,63 @@ fun AccountScreen(
         })
   }
 }
+
+/**
+ * Handles account deletion by deleting the user's profile, map user data, and Firebase account.
+ *
+ * @param password The user's password used for authentication before account deletion.
+ * @param profileViewModel The ViewModel responsible for managing the user's profile.
+ * @param mapUsersViewModel The ViewModel responsible for managing map-related user data.
+ * @param firebaseAuthViewModel The ViewModel responsible for Firebase authentication operations.
+ * @param navigationActions Used to navigate to different screens after the account deletion.
+ * @param onDeletionFailed A callback executed when the account deletion fails.
+ * @param onDeletionSuccess A callback executed when the account deletion succeeds.
+ */
+@OptIn(DelicateCoroutinesApi::class)
+private fun deleteAccount(
+    password: String,
+    profileViewModel: ProfileViewModel,
+    mapUsersViewModel: MapUsersViewModel,
+    playlistViewModel: PlaylistViewModel,
+    firebaseAuthViewModel: FirebaseAuthViewModel,
+    navigationActions: NavigationActions,
+    onDeletionFailed: (String) -> Unit,
+    onDeletionSuccess: () -> Unit
+) {
+  if (password.isEmpty()) {
+    onDeletionFailed("Please enter your password")
+    return
+  }
+
+  kotlinx.coroutines.GlobalScope.launch {
+    try {
+      // Step 1: Verify the password
+      val verificationResult = firebaseAuthViewModel.verifyPassword(password)
+      if (!verificationResult.isSuccess) {
+        withContext(Dispatchers.Main) { onDeletionFailed("Incorrect password") }
+        return@launch
+      }
+
+      // Step 2: Delete Firestore data
+      val deleteProfileResult = profileViewModel.deleteProfile()
+      val deleteMapUserResult = mapUsersViewModel.deleteMapUser()
+      val deletePlaylistsResult = playlistViewModel.deleteOwnedPlaylists()
+
+      if (!deleteProfileResult || !deleteMapUserResult || !deletePlaylistsResult) {
```

Comment:
This condition returns on a failed deletion, but the failed deletion could happen because of just the data not existing. This could happen with mapUser data and Playists data. So all data that existed will be deleted but because of a fail of deletion on a data that doesn't exist it will trigger this if condition and will not let us delete the account properly. And we will stay on the app with all data related to the account deleted except the firebase authentication data. You should find a fix to this. I recommend changing the behavior of delete function in the various viewModels.